o1vm/src/pickles/column_env.rs

```diff
@@ -34,42 +35,59 @@ pub struct ColumnEnvironment<'a, F: FftField> {
     pub domain: EvaluationDomains<F>,
 }
 
-impl<'a, F: FftField> TColumnEnvironment<'a, F, BerkeleyChallengeTerm, BerkeleyChallenges<F>>
-    for ColumnEnvironment<'a, F>
-{
-    // FIXME: do we change to the MIPS column type?
-    // We do not want to keep kimchi_msm/generic prover
-    type Column = kimchi_msm::columns::Column;
+pub fn get_all_columns() -> Vec<Column> {
```

Comment:
What about using an iterator instead?

Reply:
Do you mean like a trait object?  Otherwise we still need to pick an implementation of the iterator, and I don't see why Vec shouldn't be used simply instead.